File: public/dsa/arrays/arithmetic_sequence/optimal/arithmetic_sequence.cpp

#include <bits/stdc++.h>
using namespace std;

bool isArithmeticSequence(vector<int> &arr) {
    int n = arr.size();
    if (n == 1) {
        return true;
    }
    sort(arr.begin(), arr.end());
    int commonDiff = arr[1] - arr[0];
    for (int i = 2; i < n; i++) {
        if (arr[i] - arr[i - 1] != commonDiff) {
            return false;
        }
    }
    return true;
}

// tests-start
int main() {
    vector<int> arr = {9, 13, 5, 15, 7, 11};
    bool res = isArithmeticSequence(arr);
    cout << res << endl;
    return 0;
}
// tests-end

/*output
1
*/

File: public/dsa/arrays/arithmetic_sequence/optimal/arithmetic_sequence.py
from typing import List


class Solution:
    def isArithmeticSequence(self, arr: List[int]) -> bool:
        arr.sort()
        n = len(arr)
        cd = arr[1] - arr[0]  # common difference
        for i in range(2, n):
            if arr[i] - arr[i-1] != cd:
                return False
        return True


# tests-start
sol = Solution()
arr = [9, 13, 5, 15, 7, 11]
sol.isArithmeticSequence(arr)
# tests-end

'''output
True
'''
